File: features/favorites/src/main/java/com/vp/favorites/FavoriteViewModel.java
package com.vp.favorites;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.vp.detail.model.MovieDetail;
import com.vp.detail.utils.SharedPreference;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class FavoriteViewModel extends AndroidViewModel {
    private SharedPreference sharedPreference;

    private MutableLiveData<List<MovieDetail>> liveData = new MutableLiveData<>();

    public FavoriteViewModel(@NonNull Application application) {
        super(application);
        sharedPreference = new SharedPreference(application.getApplicationContext());
        getData();
    }


    public LiveData<List<MovieDetail>> getLiveData() {
        return liveData;
    }

    private List<MovieDetail> getData() {
        List<MovieDetail> data = new ArrayList<>();

        HashMap<String, MovieDetail> list = sharedPreference.getListId();

        if (list != null) {
            for (String i : list.keySet()) {
                data.add(list.get(i));
            }
        }
        liveData.setValue(data);

        return data;

    }


}

File: features/favorites/src/main/java/com/vp/favorites/FavoriteViewModel.kt
package com.vp.favorites

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.vp.favorite.FavoriteMovieRepository
import com.vp.favorite.model.Movie
import kotlinx.coroutines.launch
import javax.inject.Inject

class FavoriteViewModel @Inject constructor(
    private val favoriteMovieRepository: FavoriteMovieRepository
) : ViewModel() {
    private val _movies = MutableLiveData<List<Movie>>()
    val movies: LiveData<List<Movie>> = _movies

    init {
        refresh()
    }

    fun refresh() {
        viewModelScope.launch {
            val movies = favoriteMovieRepository.getFavoriteMovies()
            _movies.value = movies
        }
    }

    fun remove(id: String) {
        viewModelScope.launch {
            favoriteMovieRepository.removeFavorite(id)
            refresh()
        }
    }

    override fun onCleared() {
        super.onCleared()
        Log.d("FavoriteList", "onCleared:")
    }
}
